include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -0,0 +1,406 @@
+/**
+ * @file nsga2_impl.hpp
+ * @author Sayan Goswami
+ *
+ * Implementation of the NSGA-II algorithm. Used for multi-objective
+ * optimization problems on arbitrary functions.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more Information.
+ */
+
+#ifndef ENSMALLEN_NSGA2_NSGA2_IMPL_HPP
+#define ENSMALLEN_NSGA2_NSGA2_IMPL_HPP
+
+#include "nsga2.hpp"
+
+namespace ens {
+
+inline NSGA2::NSGA2(const size_t populationSize,
+                    const size_t maxGenerations,
+                    const double crossoverProb,
+                    const double mutationProb,
+                    const double mutationStrength,
+                    const double epsilon,
+                    const arma::vec& lowerBound,
+                    const arma::vec& upperBound) :
+    populationSize(populationSize),
+    maxGenerations(maxGenerations),
+    crossoverProb(crossoverProb),
+    mutationProb(mutationProb),
+    mutationStrength(mutationStrength),
+    epsilon(epsilon),
+    lowerBound(lowerBound),
+    upperBound(upperBound)
+{ /* Nothing to do here. */ }
+
+//! Optimize the function.
+template<typename MatType,
+         typename... ArbitraryFunctionType,
+         typename... CallbackTypes>
+typename MatType::elem_type NSGA2::Optimize(std::tuple<ArbitraryFunctionType...>& objectives,
+                                     MatType& iterate,
+                                     CallbackTypes&&... callbacks)
```

Comment:
```suggestion
typename MatType::elem_type NSGA2::Optimize(
    std::tuple<ArbitraryFunctionType...>& objectives,
    MatType& iterate,
    CallbackTypes&&... callbacks)
```
Simple style fix---the line was too long. :)

Reply:
Done. :)